### PR TITLE
Move up to fstrings

### LINKED ISSUE #1659: Move up to fstrings
[fstrings](https://twitter.com/mariatta/status/1001141610107699200)

I guess this means going up to Python 3.6 and losing Python 3.5 support.

### PR DESCRIPTION
**Pull request by DrFaustie**
Fixes #1659 

For when it's necessary to drop support for older Python Versions.

- [x] `kinto/config/__init__.py`
- [x] `kinto/schema_validation.py`
- [x] `kinto/core/scripts.py`
- [x] `kinto/core/openapi.py`
- [x] `kinto/core/__init__.py`
- [x] `kinto/core/utils.py`
- [x] `kinto/core/cache/postgresql/__init__.py`
- [x] `kinto/core/storage/memory.py`
- [x] `kinto/core/storage/exceptions.py`
- [x] `kinto/core/storage/postgresql/client.py`
- [x] `kinto/core/storage/testing.py`
- [x] `kinto/core/statsd.py`
- [x] `kinto/core/initialization.py`
- [x] `kinto/core/resource/model.py`
- [x] `kinto/core/resource/schema.py`
- [x] `kinto/core/permission/memory.py`
- [x] `kinto/core/views/heartbeat.py`
- [x] `kinto/core/views/errors.py`
- [x] `kinto/core/views/batch.py`
- [x] `kinto/core/testing.py`
- [x] `kinto/plugins/quotas/scripts.py`
- [x] `kinto/plugins/quotas/listener.py`
- [x] `kinto/plugins/openid/__init__.py`
- [x] `kinto/plugins/openid/views.py`
- [x] `kinto/plugins/default_bucket/__init__.py`
- [x] `kinto/plugins/history/listener.py`
- [x] `kinto/authorization.py`
- [x] `kinto/views/records.py`
- [x] `kinto/core/authentication.py`
- [x] `kinto/core/events.py`
- [x] `kinto/core/storage/postgresql/__init__.py`
- [x] `kinto/core/storage/postgresql/migrator.py`
- [x] `kinto/core/errors.py`
- [x] `kinto/core/resource/__init__.py`
- [x] `kinto/core/resource/viewset.py`
- [x] `kinto/core/permission/postgresql/__init__.py`
- [x] `kinto/core/authorization.py`
- [x] `kinto/plugins/accounts/scripts.py`
- [x] `kinto/plugins/accounts/__init__.py`
- [x] `kinto/plugins/accounts/views.py`
- [x] `kinto/plugins/accounts/authentication.py`
- [x] `kinto/views/permissions.py`
- [x] `kinto/views/groups.py`
- [x] `kinto/__main__.py`

# Cleanup

- [x] Add documentation.
- [x] Add tests.
- [x] Add a changelog entry.

### CODE REVIEW

**Comment by DrFaustie** [comment]
There may be some issues still. My local builds still fail on lint and functional ENV's.
https://i.imgur.com/6UrNgXI.png

There are also some lines of code that I'm unsure of on how to convert to fstrings at the moment. I'll look into those over the next few days. Code review/advice is very much appreciated

File: kinto/core/authentication.py [hunk 1]
```diff
@@ -34,6 +34,7 @@ def unauthenticated_userid(self, request):
                 return
 
             hmac_secret = settings["userid_hmac_secret"]
+            # Unsure how to use fstring with *args
```
Comment (leplatrem) [inline kinto/core/authentication.py]: I would say that `f"{credentials[0]}:{credentials[1]"` is fine. Or leave the format() call

File: kinto/core/events.py [hunk 2]
```diff
@@ -29,6 +29,7 @@ def __init__(self, payload, request):
         self.request = request
 
     def __repr__(self):
+        # Not sure what proper conversion to fstring is
```
Comment (leplatrem) [inline kinto/core/events.py]: This is probably better to leave it with format()

**Comment by leplatrem** [comment]
It looks to me! Thanks for taking the time to do this!

> Perhaps it is correct for the functional build to fail as it runs on 3.5?

This PR will drop support of Python 3.5.
The TravisCI configuration and CHANGELOG should also be updated to reflect that ;)

It looks like you should run black to format your changes (see TravisCI job py36)

File: kinto/core/events.py (hunk 2 above)
Comment (DrFaustie) [reply kinto/core/events.py]: We could do this, I think it should work

`    def __repr__(self):
        return f"<{self.__class__.__name__} {payload['action']} {payload['uri']}>"`

**Comment by DrFaustie** [comment]
It's my pleasure!

About the TravisCI config, we should update the python version but also tell Travis that we expect 3.5 to fail, correct?

**Comment by Natim** [comment]
>  tell Travis that we expect 3.5 to fail

No we should remove the travis config for py35 altogether.

**Comment by DrFaustie** [comment]
> 
> 
> Thank you for this work. To be honest I didn't expect it to be so much of an improvement. 👍 🎱

You and @leplatrem have been very kind to me and made my first experience with open source extremely welcoming. I'm very happy to have started here at Kinto 😀

**Comment by Natim** [comment]
Let's make a quick poll: https://twitter.com/

**Comment by DrFaustie** [comment]
> 
> 
> Let's make a quick poll: https://twitter.com/

I think you may need to edit your link

**Comment by Natim** [comment]
I did indeed :dancer:

**Comment by Natim** [comment]
10 days later: 
image

**Comment by DrFaustie** [comment]
Looks good!

**Comment by leplatrem** [comment]
Could you please update your branch, we'll merge ;)

**Comment by DrFaustie** [comment]
Is there anything that still needs to be done from me? 👍

**Comment by Natim** [comment]
Nope it seems good to me :+1:

**Comment by DrFaustie** [comment]
Changelog is not properly dated on 12.0.0

**Comment by glasserc** [comment]
It's still unreleased and it's marked as such, so I think it's OK, right?

**Pull request merged**